generators/heroku/index.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable consistent-return */
+const chalk = require('chalk');
+const fs = require('fs');
+const HerokuGenerator = require('generator-jhipster/generators/heroku');
+const constants = require('generator-jhipster/generators/generator-constants');
+
+module.exports = class extends HerokuGenerator {
+    constructor(args, opts) {
+        super(args, { fromBlueprint: true, ...opts }); // fromBlueprint variable is important
+
+        const jhContext = (this.jhipsterContext = this.options.jhipsterContext);
+
+        if (!jhContext) {
+            this.error(`This is a JHipster blueprint and should be used only like ${chalk.yellow('jhipster --blueprint micronaut')}`);
+        }
+
+        this.configOptions = jhContext.configOptions || {};
+    }
+
+    get initializing() {
+        return super._initializing();
+    }
+
+    get prompting() {
+        return super._prompting();
+    }
+
+    get configuring() {
+        return super._configuring();
+    }
+
+    get default() {
+        // Here we are not overriding this phase and hence its being handled by JHipster
+        const phaseFromJHipster = super._default();
+        const jhipsterMicronautDefaultPhaseSteps = {
+            copyHerokuFiles() {
+                if (this.abort) return;
+
+                const done = this.async();
+                this.log(chalk.bold('\nCreating Heroku deployment files'));
+
+                // this.template('bootstrap-heroku.yml.ejs', `${constants.SERVER_MAIN_RES_DIR}/config/bootstrap-heroku.yml`);
+                this.template('application-heroku.yml.ejs', `${constants.SERVER_MAIN_RES_DIR}/application-heroku.yml`);
+                this.template('Procfile.ejs', 'Procfile');
+                this.template('system.properties.ejs', 'system.properties');
+                if (this.buildTool === 'gradle') {
+                    this.template('heroku.gradle.ejs', 'gradle/heroku.gradle');
+                }
+                if (this.useOkta) {
+                    this.template('provision-okta-addon.sh.ejs', 'provision-okta-addon.sh');
+                    fs.appendFile('.gitignore', 'provision-okta-addon.sh', 'utf8', (err, data) => {
+                        this.log(`${chalk.yellow.bold('WARNING!')}Failed to add 'provision-okta-addon.sh' to .gitignore.'`);
+                    });
+                }
+
+                this.conflicter.resolve(err => {
```

Comment:
Thats just copied from the main generator for now. There some things (as you already discovered) which are not inline with out latest approaches. I most likely will do some refactorings here and backport them to the main generator. For example the check for heroku cli is imho too late in the process

Reply:
I don’t think this is required, this is done at yeoman-generator at conflicts step.
You should try to remove, not sure but this can prevent prettier transform to be used for example.
Marking as resolved since this was just a question.